sale_mrp_bom/models/sale_order_line.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2020 Akretion Renato Lima <renato.lima@akretion.com.br>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    bom_id = fields.Many2one(
+        comodel_name="mrp.bom",
+        string="BOM",
+        domain="[('product_tmpl_id.product_variant_ids', '=', product_id),"
```

Comment:
Is this domain sufficient? When creating a new BoM the product of the new BoM can be different to the product of the sale order line? Maybe add a constraint for this?